using-timescaledb/distributed-hypertables.md

```diff
@@ -48,32 +54,203 @@ CREATE TABLE conditions (
+   access node finalizes the result by computing the `max` over all
+   the data nodes' `max` results:
 
-### Changing the Number of Data Nodes for a Distributed Hypertable [](changing-data-nodes)
+   ```
+   SELECT max(temperature) FROM conditions;
+   ```
 
-By default, distributed hypertables will use all available data nodes when
-created. But, if you add new data nodes to the distributed database, they are
-not automatically added to existing distributed hypertables, and you need to
-explicitly attach them using [`attach_data_node`][attach_data_node].
+>:TIP: For push down to work, the settings variable
+>[`enable_partitionwise_aggregate`][partitionwise_agg_guc] must be set
+>to `TRUE` on the access node (the default is `FALSE`).
+
+Other types of push downs can include sorting operations, groupings,
+and joins. Push down for joins is currently not supported, however. To
+see how the a query is pushed down to a data node, use `EXPLAIN
+VERBOSE` on the query and inspect the query plan and the remote SQL
+statement sent to each data node.
+
+### Limitations of Query Push Down
+
+The query planner might not always be able to push down queries,
+however, or it can only push down parts of it:
+
+- If the query involves a region that saw a change in the partitioning
+  configuration (e.g., the number of space partitions increased because the
+  system was elastically expanded to include additional data nodes, and the
+  number of space partitions was correspondingly increased), then
+  a full push down might not be possible when it otherwise would
+  be. In our example distributed hypertable, the change in
+  partitioning might lead to data for the same location existing in
+  two chunks that are stored on different data nodes. The planner must
+  therefore avoid *full* push down whenever the query covers the
+  repartitioning boundary. A *partial* push down is still possible,
+  however.
+- If the query includes non-immutable functions and expressions, the
+  planner cannot push down those parts of the query since they cannot
+  be guaranteed to generate a consistent result across each data
+  node. For instance, the [`random()`][random-func] function is
+  [`VOLATILE`][volatility] since it depends on the current seed and
+  the state of the pseudo-random sequence. If the function was to be
+  pushed down to each data node, it would not generate a valid
+  pseudo-random sequence from the point of view of the access node
+  that executes the query. Another example is the
+  [`now()`][current_time] function to get the current transaction
+  time. This is a [`STABLE`][volatility] function that depends on the
+  current time zone setting on each node. TimescaleDB specifically
+  optimizes the use of `now()`, however, by constifying the function
+  on the access node, thus it is able to push down the resulting
+  timestamp to the data nodes, but this is not the case for many
+  `STABLE` functions.
```

Comment:
This seems to mixing several different concepts. Would be good to split it up into separate points, for example, one point about `now()` being optimized, one about handling of `VOLATILE` functions, and one about handling of `STABLE` functions.

Reply:
Which concepts are you thinking of? To me this is all about the ability to push down based on the volatility of a function.

Reply:
The bullet point is talking about volatile function, stable functions, and now in different ways. It's quite hard to follow and understand what is relevant. I would suggest to break out the point about "now" being optimized and put it separately. I would also suggest to talk about volatile and stable functions separately.

Reply:
volitile and stable functions are both non-immutable, so it makes sense to talk about them together. The important thing here is that they are _not_ immutable, so in that case I think it makes more sense to just say they're both non-immutable.

Reply:
I removed mentioning stable and volatile and moved the optimization for now out of the bullet list.